lib/ABI/FunctionType/Layout.cpp

```diff
@@ -813,14 +823,22 @@ Layout::Layout(const model::RawFunctionType &Function) {
     revng_assert(StackArgType.Qualifiers().empty());
     revng_assert(StackArgType.is(model::TypeKind::StructType));
 
+    const model::Type *OriginalStackType = StackArgType.UnqualifiedType().get();
+
     auto &Argument = Arguments.emplace_back();
 
+    // (WIP): I have FUD about this, so I'm leaving it out for now, let's
+    //        discuss RFT's stack argument comment during the review process.
+    //
+    // Argument.Comment = StackArgType.UnqualifiedType().get()->Comment();
+    // Argument.Name = "stack_args"; // revng-c also hard-codes this name and we
+    //                                  _really_ want to be consistent.
```

Comment:
No, we (almost) never commit generated stuff to the model.

Reply:
Yep, but this is a layout object, it's always created anew on demand and is never serialized.

Reply:
You sure? Don't we use this for conversion purposes?

Reply:
Layouts are never used for conversion - they just share a bunch of code with `CFT -> RFT` path. This line is specifically inside the `Layout` constructor though.

Reply:
Mmmmh, then why do we record this in `Layout` at all? Is there any real user?

Reply:
To make the new comment emitting thing function-type agnostic. It only relies on the layout.

When implementing it initially, I noticed that I'm duplicating layout-like logic - so I just extended it with what I was missing instead.

Reply:
For final emission we usually don't use layout, in revng-c for instance I think we have specific logic for both.

If it's not for conversion... I'd just let the emitter do ad-hoc work for `CABIFunctionType` and `RawFunctionType`. In this case, going through `Layout` doesn't seem to bring much benefits.

Reply:
> in revng-c for instance I think we have specific logic for both

There's no need to integrate that deep. Since the comment is always before the function itself, it's as simple as calling `functionComment(ForThisFunction, InThisBinary)` and just concatentating resulting string with the function decompiler emitted.

Here's literally the entirety of code on that side that handles the functions: https://rev.ng/gitlab/revng-private/revng-c/-/merge_requests/151/diffs?commit_id=934b2ef51e4457f34495f7e67e62230fb953e242#be93403bff9d784fb72272719009d4716bd81c81_1632_1635

Reply:
> In this case, going through `Layout` doesn't seem to bring much benefits.

I feel like it reduces code duplication a bit. But frankly, I'm fine with either way.

Reply:
You can factor common code, but don't go through `Layout`. `Layout` was not designed for this, let's leave it out of the equation entirely.

Reply:
I dropped changes to the layout.